src/anyio/_core/_compat.py

```diff
@@ -76,6 +76,16 @@ def __await__(self):
             yield
 
 
+class DeprecatedAwaitableList(Generic[T], List[T], DeprecatedAwaitable):
```

Comment:
Huh. Do we actually need the `Generic[T]` part here?

Reply:
Yes, it's needed for any class that requires a type argument.

Reply:
I thought if you already inherited from a GenericAlias it worked:

https://mypy-play.net/?mypy=latest&python=3.9&gist=fdc3ca168923e66527952336ccf7f320

Reply:
I couldn've sworn that I tried this already. I guess I left out the type argument.